webapp/test/list.spec.js

```diff
@@ -0,0 +1,81 @@
+import { mount } from '@vue/test-utils';
+import list from '../src/components/list.vue';
+
+let mocks, propsData, wrapper
+const items = [
+  { id: "0", editMode: false, text: "Take out the trash" },
+  { id: "1", editMode: false, text: "Get A+ in SDF" },
+  { id: "2", editMode: false, text: "Trashtalk the vue 'Getting started' guide" },
+  { id: "3", editMode: false, text: "git commit -m 'this'" }
+];
+
+//function for creating the wrapper in tests that need them
+const Wrapper = () => {
+  return mount(list, {
+    mocks,
+    propsData
+  }
+  )
+}
+
+describe('list.vue', () => {
+  beforeEach(() => {
+    propsData = { items: items };
+    mocks = {
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue(32)
+      },
+      items: {}
+    };
+    wrapper = Wrapper()
+  });
+
+  it('renders a list', () => {
+    expect(wrapper.contains('div')).toBe(true);
+    expect(wrapper.findAll('p')).toHaveLength(4)
+  });
+
+  describe('when the add entry button is pressed', () => {
+    beforeEach(() => {
+      wrapper = Wrapper()
+    })
+
+    it('emits an createEntry event', () => {
+      const buttons = wrapper.findAll('button');
+      const addButton = buttons.at((buttons.length - 1));
+      addButton.trigger('click');
+      expect(wrapper.emitted('createEntry'));
+    });
+    //Test fails due to lack of understanding of apollo mockups in jest
+    // it('attempts to call an apollo mutation', () => {
+    //   expect(mutate).toBeCalled();
```

Comment:
You have access to a local variable `mocks` here, declared on line 4.
```suggestion
    expect(mocks.$apollo.mutate).toHaveBeenCalled();
```